lib/assets/javascripts/turbograft/remote.coffee

```diff
@@ -73,6 +70,20 @@ class TurboGraft.Remote
 
   uriEncodeForm: (form) ->
     formData = ""
+    @_iterateOverFormInputs form, (input) =>
+      formData = @formAppend(formData, input.name, input.value)
+    formData
+
+  nativeEncodeForm: (form) ->
+    formData = new FormData
+    @_iterateOverFormInputs form, (input) =>
+      if input.type == 'file'
+        formData.append(input.name, input.files[0]) # <- just a guess
```

Comment:
We'll also need to serialized all files in the input, not just the first

On Thursday, February 5, 2015, Anthony Cameron <notifications@github.com>
wrote:

> In lib/assets/javascripts/turbograft/remote.coffee
> <https://github.com/Shopify/turbograft/pull/67#discussion_r24169080>:
>
> > @@ -73,6 +70,20 @@ class TurboGraft.Remote
> >
> >    uriEncodeForm: (form) ->
> >      formData = ""
> > +    @_iterateOverFormInputs form, (input) =>
> > +      formData = @formAppend(formData, input.name, input.value)
> > +    formData
> > +
> > +  nativeEncodeForm: (form) ->
> > +    formData = new FormData
> > +    @_iterateOverFormInputs form, (input) =>
> > +      if input.type == 'file'
> > +        formData.append(input.name, input.files[0]) # <- just a guess
>
> I think it might be input.files though
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/Shopify/turbograft/pull/67/files#r24169080>.
>